File: src/main/java/frc/robot/subsystems/NavX.java
package frc.robot.subsystems;

import edu.wpi.first.wpilibj2.command.SubsystemBase;

import com.kauailabs.navx.frc.AHRS;

public class NavX extends SubsystemBase{
    private final AHRS navx = new AHRS();

    public NavX(){
        reset();
    }

    public void reset(){
        navx.reset();
    }

    public double getHeading(){ //in degrees
        return navx.getAngle();
    }

    public boolean isConnected(){
        return navx.isConnected();
    }

    @Override
    public void periodic() {}
}

File: src/main/java/frc/robot/subsystems/NavX.kt
package frc.robot.subsystems

import com.kauailabs.navx.frc.AHRS
import edu.wpi.first.wpilibj2.command.SubsystemBase

class NavX : SubsystemBase() {
    private val navx = AHRS()

    init {
        reset()
    }

    fun reset() {
        navx.reset()
    }

    //in degrees
    val heading: Double
        get() = navx.angle

    val isConnected: Boolean
        get() = navx.isConnected

    override fun periodic() {}
}